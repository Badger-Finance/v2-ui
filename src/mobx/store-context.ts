import { RootStore } from './store';
import { createContext } from 'react';

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;
