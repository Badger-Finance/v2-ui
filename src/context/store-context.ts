import { createContext } from "react";
import { RootStore } from "../mobx/store";

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;