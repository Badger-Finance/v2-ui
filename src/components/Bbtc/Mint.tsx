import React, { useContext, useState } from 'react';

import { Container, Button, Typography } from '@material-ui/core';

import { useStyles, debounce } from './common';
import BigNumber from 'bignumber.js';
import { Tokens } from './Tokens';

import { TokenModel } from './model';

import { StoreContext } from '../../mobx/store-context';

export const Mint = (): any => {
	const store = useContext(StoreContext);
	const classes = useStyles();

	const tokens = store.bbtcStore.tokens;
	const bBTC = store.bbtcStore.bBTC;

	let inputRef: any;

	const [selectedToken, setSelectedToken] = useState<TokenModel>(tokens[0]);
	const [inputAmount, setInputAmount] = useState<number | string>('');
	const _debouncedSetInputAmount = debounce(1000, (val) => {
		setInputAmount(val);
	});

	const handleInputAmount = (event: any) => {
		const nextValue = event?.target?.value;
		_debouncedSetInputAmount(nextValue);
	};

	const handleTokenSelection = (event: any) => {
		const token = tokens.find((token) => token.symbol === event?.target?.value);
		setSelectedToken(token || tokens[0]);
	};

	const useMaxBalance = () => {
		setInputAmount((inputRef.value = selectedToken.balance.toString(10)));
	};
	const handleMintClick = () => {
		store.bbtcStore.mint(selectedToken, selectedToken.scale(new BigNumber(inputAmount)));
	};

	return (
		<Container className={classes.root} maxWidth="lg">
			<div className={classes.outerWrapper}>
				<Typography variant="caption" className={classes.balance}>
					Available {selectedToken.symbol}: {selectedToken.formattedBalance}
				</Typography>
				<div className={classes.inputWrapper}>
					<Tokens tokens={tokens} default={selectedToken.symbol} onTokenSelect={handleTokenSelection} />
					<input
						className={classes.unstylishInput + ' unstylish-input'}
						onChange={handleInputAmount}
						type="number"
						ref={(ref) => (inputRef = ref)}
						min="0"
						placeholder="0.0"
					/>

					<Button className={classes.btnMax} variant="outlined" onClick={useMaxBalance}>
						max
					</Button>
				</div>
			</div>
			<div className={classes.outerWrapper}>
				<svg
					className={classes.arrowDown}
					width="13"
					height="18"
					viewBox="0 0 13 18"
					fill="none"
					xmlns="http://www.w3.org/2000/svg"
				>
					<path
						fillRule="evenodd"
						clipRule="evenodd"
						d="M6.5 15.2138L11.6295 10L13 11.3931L6.5 18L-6.08938e-08 11.3931L1.37054 10L6.5 15.2138Z"
						fill="white"
					/>
					<line x1="6.5" y1="16" x2="6.5" y2="4.37114e-08" stroke="white" strokeWidth="2" />
				</svg>
			</div>
			<div className={classes.outerWrapper}>
				<Typography variant="caption" className={classes.balance}>
					Available {bBTC.symbol}: {bBTC.formattedBalance}
				</Typography>
				<div className={classes.inputWrapper}>
					<div className={classes.token}>
						<img src={bBTC.icon} className={classes.tokenIcon} alt={bBTC.name} />
						<Typography className={classes.tokenLabel} variant="body1">
							{bBTC.symbol}
						</Typography>
					</div>
					<input
						className={classes.unstylishInput + ' unstylish-input'}
						type="number"
						min="0"
						value={inputAmount}
						placeholder="0.0"
						readOnly
					/>
				</div>
			</div>
			<div className={classes.outerWrapper}>
				<div className={classes.summaryWrapper}>
					<div className={classes.summaryRow}>
						<Typography variant="subtitle1">Current Conversion Rate: </Typography>
						<Typography variant="subtitle1">
							1{selectedToken.symbol}: 1{bBTC.symbol}
						</Typography>
					</div>
					<div className={classes.summaryRow}>
						<Typography variant="subtitle1">Fees: </Typography>
						<Typography variant="subtitle1">0.002 {bBTC.symbol}</Typography>
					</div>
				</div>
			</div>
			<div className={classes.outerWrapper}>
				<Button size="large" variant="contained" color="primary" onClick={handleMintClick}>
					MINT
				</Button>
			</div>
		</Container>
	);
};
