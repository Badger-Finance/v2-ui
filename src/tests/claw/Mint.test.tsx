import React from 'react';
import BigNumber from 'bignumber.js';
import '@testing-library/jest-dom';
import { customRender, fireEvent, screen, within } from '../Utils';
import { StoreProvider } from '../../mobx/store-context';
import store from '../../mobx/store';
import ClawStore from '../../mobx/stores/claw/clawStore';
import Mint from '../../components/Claw/Mint';
import * as MintHooks from '../../components/Claw/Mint/mint.hooks';
import {
	mockClawsByCollaterals,
	mockCollaterals,
	mockSyntheticData,
	mockSyntheticDataByEmp,
	mockTokens,
} from './claw.mocks';

jest.spyOn(ClawStore.prototype, 'fetchSyntheticsData');
jest.spyOn(ClawStore.prototype, 'fetchSponsorData');

describe('Claw - Mint', () => {
	beforeAll(() => {
		store.wallet.connectedAddress = '0x1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a';
		store.contracts.tokens = mockTokens;
		store.claw.collaterals = mockCollaterals;
		store.claw.syntheticsData = mockSyntheticData;
		store.claw.syntheticsDataByEMP = mockSyntheticDataByEmp;
		store.claw.clawsByCollateral = mockClawsByCollaterals;
	});

	test('matches snapshot', () => {
		const { container } = customRender(
			<StoreProvider value={store}>
				<Mint />
			</StoreProvider>,
		);
		expect(container).toMatchSnapshot();
	});

	describe('Collateral Token', () => {
		beforeEach(() =>
			customRender(
				<StoreProvider value={store}>
					<Mint />
				</StoreProvider>,
			),
		);

		describe('when a collateral token is not selected', () => {
			test('input amount is disabled', () => {
				const [collateralAmount] = screen.getAllByRole('textbox');
				expect(collateralAmount).toBeDisabled();
			});

			test('percentage buttons are disabled', () => {
				const [fiftyPercentCollateralButton] = screen.getAllByRole('button', {
					name: '50%',
				});
				expect(fiftyPercentCollateralButton).toBeDisabled();
			});

			test('action button is disabled and with "Select a Collateral Token" as text', () => {
				expect(screen.getByRole('button', { name: 'Select a Collateral Token' })).toBeDisabled();
			});
		});

		describe('when a collateral token is selected', () => {
			beforeEach(() => {
				const collateralTokenSelector = screen.getByRole('button', { name: 'Select Token' });
				fireEvent.mouseDown(collateralTokenSelector);
				const collateralsListBox = within(screen.getByRole('listbox'));
				fireEvent.click(collateralsListBox.getByText('bBADGER'));
			});

			test('token name is displayed in the select', () => {
				const [collateralAmountInput] = screen.getAllByRole('textbox');
				expect(
					screen.getByRole('button', {
						name: 'bBADGER',
					}),
				).toBeInTheDocument();
				expect(collateralAmountInput).toBeEnabled();
			});

			test('token balance is displayed', () => {
				expect(screen.getByText('522.378430153821121982')).toBeInTheDocument();
			});

			test('token mint amount can be changed', () => {
				const [collateralAmountInput] = screen.getAllByRole('textbox');
				fireEvent.change(collateralAmountInput, { target: { value: '100.00' } });
				expect(screen.getByDisplayValue('100.00')).toBeInTheDocument();
			});

			test('token balance percentage can be applied', () => {
				const [fiftyPercentCollateralButton] = screen.getAllByRole('button', {
					name: '50%',
				});

				expect(fiftyPercentCollateralButton).toBeEnabled();
				fireEvent.click(fiftyPercentCollateralButton);
				expect(screen.getByDisplayValue('261.189215076910560991')).toBeInTheDocument();
			});

			test('action button is disabled on amounts bigger that collateral token balance', () => {
				const [collateralAmountInput] = screen.getAllByRole('textbox');
				fireEvent.change(collateralAmountInput, { target: { value: '1044.00' } });
				expect(screen.getByDisplayValue('1044.00')).toBeInTheDocument();
				expect(screen.getByRole('button', { name: 'Insufficient bBADGER balance' })).toBeDisabled();
			});

			test('action button is disabled and with "Enter collateral amount" as text', () => {
				expect(screen.getByRole('button', { name: 'Enter collateral amount' })).toBeDisabled();
			});
		});
	});

	describe('Synthetic Token', () => {
		beforeEach(() => {
			jest.clearAllMocks();
			jest.spyOn(MintHooks, 'useMaxClaw').mockReturnValue(new BigNumber('200000000000000000000'));
			customRender(
				<StoreProvider value={store}>
					<Mint />
				</StoreProvider>,
			);
		});

		describe('when a collateral token is not selected', () => {
			test('amount input is disabled', () => {
				expect(screen.getByRole('button', { name: 'Select CLAW' })).toHaveAttribute('aria-disabled');
			});
			test('percentage buttons are disabled', () => {
				const [, fiftyPercentSyntheticButton] = screen.getAllByRole('button', {
					name: '50%',
				});
				expect(fiftyPercentSyntheticButton).toBeDisabled();
			});
		});

		describe('when a collateral token is selected', () => {
			beforeEach(() => {
				const collateralTokenSelector = screen.getByRole('button', { name: 'Select Token' });
				const [collateralAmountInput] = screen.getAllByRole('textbox');
				fireEvent.mouseDown(collateralTokenSelector);
				const collateralsListBox = within(screen.getByRole('listbox'));
				fireEvent.click(collateralsListBox.getByText('bBADGER'));
				fireEvent.change(collateralAmountInput, { target: { value: '100' } });
			});

			test('synthetic token select is enabled', () => {
				expect(screen.getByRole('button', { name: 'Select CLAW' })).toBeEnabled();
			});

			describe('when a synthetic token is selected too', () => {
				beforeEach(() => {
					const syntheticSelector = screen.getByRole('button', { name: 'Select CLAW' });
					fireEvent.mouseDown(syntheticSelector);
					const collateralsListBox = within(screen.getByRole('listbox'));
					fireEvent.click(collateralsListBox.getByText('USD/bBadger 5-29'));
				});

				test('selected synthetic name is displayed', () => {
					const [, syntheticAmountInput] = screen.getAllByRole('textbox');
					expect(
						screen.getByRole('button', {
							name: 'USD/bBadger 5-29',
						}),
					).toBeInTheDocument();
					expect(syntheticAmountInput).toBeEnabled();
				});

				test('correct max CLAW amount is displayed', () => {
					expect(screen.getByText('200')).toBeInTheDocument();
				});

				test('synthetic amount can be changed', () => {
					const [, syntheticAmountInput] = screen.getAllByRole('textbox');
					fireEvent.change(syntheticAmountInput, { target: { value: '150.00' } });
					expect(screen.getByDisplayValue('150.00')).toBeInTheDocument();
				});

				test('synthetic percentage can be applied', () => {
					const [, fiftyPercentSyntheticButton] = screen.getAllByRole('button', {
						name: '50%',
					});
					expect(fiftyPercentSyntheticButton).toBeEnabled();
					fireEvent.click(fiftyPercentSyntheticButton);
					expect(screen.getByDisplayValue('100')).toBeInTheDocument();
				});

				test('action button is disabled on amounts less than the minimum mint amount', () => {
					const [, syntheticAmountInput] = screen.getAllByRole('textbox');
					fireEvent.change(syntheticAmountInput, { target: { value: '10.00' } });
					expect(screen.getByDisplayValue('10.00')).toBeInTheDocument();
					expect(screen.getByRole('button', { name: 'Insufficient input CLAW amount' })).toBeDisabled();
				});

				test('action button is enabled', () => {
					const [, syntheticAmountInput] = screen.getAllByRole('textbox');
					fireEvent.change(syntheticAmountInput, { target: { value: '150.00' } });
					expect(screen.getByDisplayValue('150.00')).toBeInTheDocument();
					expect(screen.getByRole('button', { name: 'MINT' })).toBeEnabled();
				});
			});
		});
	});
});
