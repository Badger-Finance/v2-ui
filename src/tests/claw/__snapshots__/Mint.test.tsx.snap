// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claw - Mint matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root-7 MuiGrid-container-8"
  >
    <div
      class="MuiGrid-root-7 MuiBox-root-110 MuiBox-root-111 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
    >
      <div
        class="MuiGrid-root-7 MuiGrid-container-8"
      >
        <div
          class="MuiGrid-root-7 MuiBox-root-110 MuiBox-root-114 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
        >
          <div
            class="MuiGrid-root-7 MuiGrid-container-8 MuiGrid-spacing-xs-1-30"
          >
            <div
              class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-6-61"
            >
              <p
                class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
              >
                Collateral
              </p>
            </div>
            <div
              class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-6-61"
            >
              <div
                class="MuiGrid-root-7 makeStyles-availableText-115 MuiGrid-container-8 MuiGrid-spacing-xs-1-30"
              >
                <div
                  class="MuiGrid-root-7 MuiGrid-item-9"
                >
                  <p
                    class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
                  />
                </div>
                <div
                  class="MuiGrid-root-7 makeStyles-balanceText-116 MuiGrid-item-9"
                >
                  <p
                    class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextPrimary-142"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
        >
          <div
            class="MuiGrid-root-7 makeStyles-border-112 MuiBox-root-110 MuiBox-root-147 MuiGrid-container-8 MuiGrid-align-items-xs-center-16 MuiGrid-align-content-xs-center-20"
          >
            <div
              class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-8-63 MuiGrid-grid-lg-9-92"
            >
              <div
                class="MuiGrid-root-7 makeStyles-selectContainer-113 MuiGrid-container-8 MuiGrid-spacing-xs-2-31 MuiGrid-align-items-xs-center-16"
              >
                <div
                  class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-true-55"
                >
                  <div
                    class="MuiFormControl-root-149 makeStyles-margin-148"
                  >
                    <div
                      class="MuiInputBase-root-164 MuiInputBase-formControl-165"
                    >
                      <div
                        aria-haspopup="listbox"
                        class="MuiSelect-root-153 MuiSelect-select-154 MuiSelect-selectMenu-157 MuiInputBase-input-175"
                        role="button"
                        tabindex="0"
                      >
                        Select Token
                      </div>
                      <input
                        aria-hidden="true"
                        class="MuiSelect-nativeInput-163"
                        placeholder="Select Token"
                        tabindex="-1"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root-182 MuiSelect-icon-159"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-true-55"
                >
                  <div
                    class="MuiInputBase-root-164 makeStyles-input-195 MuiInputBase-disabled-167 MuiInputBase-fullWidth-174"
                  >
                    <input
                      class="MuiInputBase-input-175 MuiInputBase-disabled-167"
                      disabled=""
                      pattern="^[0-9]*[.,]?[0-9]*$"
                      placeholder="0.00"
                      type="tel"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-4-59 MuiGrid-grid-lg-3-86"
            >
              <div
                class="MuiGrid-root-7 MuiGrid-container-8 MuiGrid-justify-xs-flex-end-26"
              >
                <div
                  aria-label="text button group"
                  class="MuiButtonGroup-root-197"
                  role="group"
                >
                  <button
                    class="MuiButtonBase-root-250 MuiButton-root-221 MuiButton-text-223 MuiButtonGroup-grouped-203 MuiButtonGroup-groupedHorizontal-204 MuiButtonGroup-groupedText-206 MuiButtonGroup-groupedTextHorizontal-207 MuiButtonGroup-groupedText-206 MuiButtonGroup-disabled-200 makeStyles-button-196 MuiButton-textSizeSmall-236 MuiButton-sizeSmall-242 MuiButton-disableElevation-232 MuiButton-disabled-234 MuiButtonBase-disabled-251"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-222"
                    >
                      25
                      %
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root-250 MuiButton-root-221 MuiButton-text-223 MuiButtonGroup-grouped-203 MuiButtonGroup-groupedHorizontal-204 MuiButtonGroup-groupedText-206 MuiButtonGroup-groupedTextHorizontal-207 MuiButtonGroup-groupedText-206 MuiButtonGroup-disabled-200 makeStyles-button-196 MuiButton-textSizeSmall-236 MuiButton-sizeSmall-242 MuiButton-disableElevation-232 MuiButton-disabled-234 MuiButtonBase-disabled-251"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-222"
                    >
                      50
                      %
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root-250 MuiButton-root-221 MuiButton-text-223 MuiButtonGroup-grouped-203 MuiButtonGroup-groupedHorizontal-204 MuiButtonGroup-groupedText-206 MuiButtonGroup-groupedTextHorizontal-207 MuiButtonGroup-groupedText-206 MuiButtonGroup-disabled-200 makeStyles-button-196 MuiButton-textSizeSmall-236 MuiButton-sizeSmall-242 MuiButton-disableElevation-232 MuiButton-disabled-234 MuiButtonBase-disabled-251"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-222"
                    >
                      75
                      %
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root-250 MuiButton-root-221 MuiButton-text-223 MuiButtonGroup-grouped-203 MuiButtonGroup-groupedHorizontal-204 MuiButtonGroup-groupedText-206 MuiButtonGroup-groupedTextHorizontal-207 MuiButtonGroup-groupedText-206 MuiButtonGroup-disabled-200 makeStyles-button-196 MuiButton-textSizeSmall-236 MuiButton-sizeSmall-242 MuiButton-disableElevation-232 MuiButton-disabled-234 MuiButtonBase-disabled-251"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-222"
                    >
                      100
                      %
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
    >
      <div
        class="MuiGrid-root-7 MuiGrid-container-8"
      >
        <div
          class="MuiGrid-root-7 MuiBox-root-110 MuiBox-root-253 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
        >
          <div
            class="MuiGrid-root-7 MuiGrid-container-8 MuiGrid-spacing-xs-1-30"
          >
            <div
              class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-6-61"
            >
              <p
                class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
              >
                Mintable
              </p>
            </div>
            <div
              class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-6-61"
            >
              <div
                class="MuiGrid-root-7 makeStyles-availableText-115 MuiGrid-container-8 MuiGrid-spacing-xs-1-30"
              >
                <div
                  class="MuiGrid-root-7 MuiGrid-item-9"
                >
                  <p
                    class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
                  />
                </div>
                <div
                  class="MuiGrid-root-7 makeStyles-balanceText-116 MuiGrid-item-9"
                >
                  <p
                    class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextPrimary-142"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
        >
          <div
            class="MuiGrid-root-7 makeStyles-border-112 MuiBox-root-110 MuiBox-root-254 MuiGrid-container-8 MuiGrid-align-items-xs-center-16 MuiGrid-align-content-xs-center-20"
          >
            <div
              class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-8-63 MuiGrid-grid-lg-9-92"
            >
              <div
                class="MuiGrid-root-7 makeStyles-selectContainer-113 MuiGrid-container-8 MuiGrid-spacing-xs-2-31 MuiGrid-align-items-xs-center-16"
              >
                <div
                  class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-true-55"
                >
                  <div
                    class="MuiFormControl-root-149 makeStyles-margin-148"
                  >
                    <div
                      class="MuiInputBase-root-164 MuiInputBase-disabled-167 MuiInputBase-formControl-165"
                    >
                      <div
                        aria-disabled="true"
                        aria-haspopup="listbox"
                        class="MuiSelect-root-153 MuiSelect-select-154 MuiSelect-selectMenu-157 MuiInputBase-input-175 MuiInputBase-disabled-167 MuiSelect-disabled-158"
                        role="button"
                      >
                        Select CLAW
                      </div>
                      <input
                        aria-hidden="true"
                        class="MuiSelect-nativeInput-163"
                        placeholder="Select CLAW"
                        tabindex="-1"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root-182 MuiSelect-icon-159 MuiSelect-disabled-158"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-true-55"
                >
                  <div
                    class="MuiInputBase-root-164 makeStyles-input-195 MuiInputBase-disabled-167 MuiInputBase-fullWidth-174"
                  >
                    <input
                      class="MuiInputBase-input-175 MuiInputBase-disabled-167"
                      disabled=""
                      pattern="^[0-9]*[.,]?[0-9]*$"
                      placeholder="0.00"
                      type="tel"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53 MuiGrid-grid-sm-4-59 MuiGrid-grid-lg-3-86"
            >
              <div
                class="MuiGrid-root-7 MuiGrid-container-8 MuiGrid-justify-xs-flex-end-26"
              >
                <div
                  aria-label="text button group"
                  class="MuiButtonGroup-root-197"
                  role="group"
                >
                  <button
                    class="MuiButtonBase-root-250 MuiButton-root-221 MuiButton-text-223 MuiButtonGroup-grouped-203 MuiButtonGroup-groupedHorizontal-204 MuiButtonGroup-groupedText-206 MuiButtonGroup-groupedTextHorizontal-207 MuiButtonGroup-groupedText-206 MuiButtonGroup-disabled-200 makeStyles-button-196 MuiButton-textSizeSmall-236 MuiButton-sizeSmall-242 MuiButton-disableElevation-232 MuiButton-disabled-234 MuiButtonBase-disabled-251"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-222"
                    >
                      25
                      %
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root-250 MuiButton-root-221 MuiButton-text-223 MuiButtonGroup-grouped-203 MuiButtonGroup-groupedHorizontal-204 MuiButtonGroup-groupedText-206 MuiButtonGroup-groupedTextHorizontal-207 MuiButtonGroup-groupedText-206 MuiButtonGroup-disabled-200 makeStyles-button-196 MuiButton-textSizeSmall-236 MuiButton-sizeSmall-242 MuiButton-disableElevation-232 MuiButton-disabled-234 MuiButtonBase-disabled-251"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-222"
                    >
                      50
                      %
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root-250 MuiButton-root-221 MuiButton-text-223 MuiButtonGroup-grouped-203 MuiButtonGroup-groupedHorizontal-204 MuiButtonGroup-groupedText-206 MuiButtonGroup-groupedTextHorizontal-207 MuiButtonGroup-groupedText-206 MuiButtonGroup-disabled-200 makeStyles-button-196 MuiButton-textSizeSmall-236 MuiButton-sizeSmall-242 MuiButton-disableElevation-232 MuiButton-disabled-234 MuiButtonBase-disabled-251"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-222"
                    >
                      75
                      %
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root-250 MuiButton-root-221 MuiButton-text-223 MuiButtonGroup-grouped-203 MuiButtonGroup-groupedHorizontal-204 MuiButtonGroup-groupedText-206 MuiButtonGroup-groupedTextHorizontal-207 MuiButtonGroup-groupedText-206 MuiButtonGroup-disabled-200 makeStyles-button-196 MuiButton-textSizeSmall-236 MuiButton-sizeSmall-242 MuiButton-disableElevation-232 MuiButton-disabled-234 MuiButtonBase-disabled-251"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-222"
                    >
                      100
                      %
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
    >
      <div
        class="MuiGrid-root-7 makeStyles-details-4 MuiGrid-container-8"
      >
        <div
          class="MuiGrid-root-7 makeStyles-root-255 MuiGrid-container-8"
        >
          <div
            class="MuiGrid-root-7 makeStyles-row-256 MuiGrid-container-8"
          >
            <div
              class="MuiGrid-root-7 makeStyles-detailName-257 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <p
                class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
              >
                Liquidation Price
              </p>
            </div>
            <div
              class="MuiGrid-root-7 makeStyles-placeholder-259 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <div
                class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
              >
                <p
                  class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextPrimary-142"
                >
                  -
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-7 makeStyles-row-256 MuiGrid-container-8"
          >
            <div
              class="MuiGrid-root-7 makeStyles-detailName-257 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <p
                class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
              >
                Collateral Ratio - Global
              </p>
            </div>
            <div
              class="MuiGrid-root-7 makeStyles-placeholder-259 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <div
                class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
              >
                <p
                  class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextPrimary-142"
                >
                  -
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-7 makeStyles-row-256 MuiGrid-container-8"
          >
            <div
              class="MuiGrid-root-7 makeStyles-detailName-257 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <p
                class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
              >
                Collateral Ratio - Minimum
              </p>
            </div>
            <div
              class="MuiGrid-root-7 makeStyles-placeholder-259 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <div
                class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
              >
                <p
                  class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextPrimary-142"
                >
                  -
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-7 makeStyles-row-256 MuiGrid-container-8"
          >
            <div
              class="MuiGrid-root-7 makeStyles-detailName-257 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <p
                class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
              >
                Collateral Ratio - Current
              </p>
            </div>
            <div
              class="MuiGrid-root-7 makeStyles-placeholder-259 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <div
                class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
              >
                <p
                  class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextPrimary-142"
                >
                  -
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-7 makeStyles-row-256 MuiGrid-container-8"
          >
            <div
              class="MuiGrid-root-7 makeStyles-detailName-257 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <p
                class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
              >
                Expiration
              </p>
            </div>
            <div
              class="MuiGrid-root-7 makeStyles-placeholder-259 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <div
                class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
              >
                <p
                  class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextPrimary-142"
                >
                  -
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-7 makeStyles-row-256 MuiGrid-container-8"
          >
            <div
              class="MuiGrid-root-7 makeStyles-detailName-257 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <p
                class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextSecondary-143"
              >
                Minimum Mint
              </p>
            </div>
            <div
              class="MuiGrid-root-7 makeStyles-placeholder-259 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-grid-xs-6-47"
            >
              <div
                class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
              >
                <p
                  class="MuiTypography-root-117 MuiTypography-body2-118 MuiTypography-colorTextPrimary-142"
                >
                  -
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-7 MuiGrid-item-9 MuiGrid-grid-xs-12-53"
    >
      <div
        class="MuiGrid-root-7 MuiGrid-container-8"
      >
        <button
          class="MuiButtonBase-root-250 MuiButton-root-221 MuiButton-contained-229 makeStyles-button-260 MuiButton-containedPrimary-230 MuiButton-containedSizeLarge-241 MuiButton-sizeLarge-243 MuiButton-disabled-234 MuiButtonBase-disabled-251"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label-222"
          >
            Select a Collateral Token
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
